### PR TITLE
test: add carousel native tests

### PR DESCRIPTION
**Pull request by anuraghazra**

### CODE REVIEW

File: packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        <CarouselItem>
+          <TestimonialCard />
+        </CarouselItem>
+      </Carousel>,
+    );
+
+    expect(queryAllByTestId('active-slide')[0]).toHaveTextContent('0');
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 0);
+
+    await act(() => {
```
Comment (snitin315) [inline packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx:195]: Screenshot 2023-08-11 at 7 58 53 AM

```suggestion
    await act(async () => {
```

File: packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+      jest.advanceTimersByTime(6000);
+    });
+    expect(onChange).toHaveBeenLastCalledWith(1);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 1);
+
+    fireEvent(getByLabelText('My Carousel'), 'onScrollBeginDrag');
+
+    await act(() => {
```
Comment (snitin315) [inline packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx:203]:
```suggestion
    await act(async () => {
```

File: packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+      jest.advanceTimersByTime(6000);
+    });
+
+    expect(onChange).toHaveBeenLastCalledWith(1);
+    expect(getByRole('tab', { selected: true })).toHaveProp('slideIndex', 1);
+
+    fireEvent(getByLabelText('My Carousel'), 'onScrollEndDrag');
+
+    await act(() => {
```
Comment (snitin315) [inline packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx:212]:
```suggestion
    await act(async () => {
```

File: packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx (hunk 1 above)
Comment (anuraghazra) [reply packages/blade/src/components/Carousel/__tests__/Carousel.native.test.tsx]: Done.

**Review by snitin315 — APPROVED** [review]

**Pull request merged**